lib/bq_stream/bq_stream.rb

```diff
@@ -14,86 +14,85 @@ module BqStream
   define_setting :back_date, nil
   define_setting :batch_size, 1000
 
-  def self.log
-    @bq_log ||= Logger.new("#{File.expand_path('')}/log/bq_stream_errors.log",
-                           File::WRONLY | File::APPEND)
-  end
+  class << self
+    attr_accessor :logger
 
-  def self.logger
-    @bq_logger ||=
-      Logger.new("#{File.expand_path('')}/log/bq_stream_errors.log",
-                 File::WRONLY | File::APPEND)
-  end
+    def self.log(type, message)
+      return unless logger
+      type = :info unless %i[unknown fatal error warn info debug].include?(type)
+      logger.send(type, message)
+    end
 
-  def self.create_bq_writer
-    require 'big_query'
-    opts = {}
-    opts['client_id']     = client_id
-    opts['service_email'] = service_email
-    opts['key']           = key
-    opts['project_id']    = project_id
-    opts['dataset']       = dataset
-    @bq_writer = BigQuery::Client.new(opts)
-  end
+    def create_bq_writer
+      require 'big_query'
+      opts = {}
+      opts['client_id']     = client_id
+      opts['service_email'] = service_email
+      opts['key']           = key
+      opts['project_id']    = project_id
+      opts['dataset']       = dataset
+      @bq_writer = BigQuery::Client.new(opts)
+    end
 
-  def self.config_initialized
-    QueuedItem.build_table
-    OldestRecord.build_table
-    create_bq_writer
-    create_bq_dataset unless @bq_writer.datasets_formatted.include?(dataset)
-    create_bq_table unless @bq_writer.tables_formatted.include?(bq_table_name)
-    initialize_old_records
-  end
+    def config_initialized
+      QueuedItem.build_table
+      OldestRecord.build_table
+      create_bq_writer
+      create_bq_dataset unless @bq_writer.datasets_formatted.include?(dataset)
+      create_bq_table unless @bq_writer.tables_formatted.include?(bq_table_name)
+      initialize_old_records
+    end
 
-  def self.initialize_old_records
-    # logger.info "#{Time.now}: $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$ Start Initialize Old Records $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$"
-    old_records = @bq_writer.query('SELECT table_name, attr, min(updated_at) '\
-                                   'as bq_earliest_update FROM '\
-                                   "[#{project_id}:#{dataset}.#{bq_table_name}] "\
-                                   'GROUP BY table_name, attr')
-    old_records['rows'].each do |r|
-      table = r['f'][0]['v']
-      trait = r['f'][1]['v']
-      unless trait.nil?
-        rec = OldestRecord.find_or_create_by(table_name: table, attr: trait)
-        rec.update(bq_earliest_update: Time.at(r['f'][2]['v'].to_f))
-      end
-    end if old_records['rows']
-    # logger.info "#{Time.now}: $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$ End Initialize Old Records $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$"
-  end
+    def initialize_old_records
+      logger.info "#{Time.now}: $$$$$$$$$$ Start Init Old Records $$$$$$$$$$"
```

Comment:
So this is going to break if we don't set a logger. The point of the method we created up above to is optionally output to logs if the logger exists.
```ruby
log(:info, "#{Time.now}: $$$$$$$$$$ Start Init Old Records $$$$$$$$$$")`
```